container/lxd/initialisation_test.go

```diff
@@ -72,3 +72,97 @@ func (s *InitialiserSuite) TestNoSeriesPackages(c *gc.C) {
 		paccmder.InstallCmd("lxd"),
 	})
 }
+
+func (s *InitialiserSuite) TestEditLXDBridgeFile(c *gc.C) {
+	input := `# WARNING: Don't modify this file by hand, it is generated by debconf!
+# To update those values, please run "dpkg-reconfigure lxd"
+
+# Whether to setup a new bridge
+USE_LXD_BRIDGE="true"
+EXISTING_BRIDGE=""
+
+# Bridge name
+LXD_BRIDGE="lxdbr0"
+
+# dnsmasq configuration path
+LXD_CONFILE=""
+
+# dnsmasq domain
+LXD_DOMAIN="lxd"
+
+# IPv4
+LXD_IPV4_ADDR="10.0.4.1"
+LXD_IPV4_NETMASK="255.255.255.0"
+LXD_IPV4_NETWORK="10.0.4.1/24"
+LXD_IPV4_DHCP_RANGE="10.0.4.2,10.0.4.100"
+LXD_IPV4_DHCP_MAX="50"
+LXD_IPV4_NAT="true"
+
+# IPv6
+LXD_IPV6_ADDR="2001:470:b2b5:9999::1"
+LXD_IPV6_MASK="64"
+LXD_IPV6_NETWORK="2001:470:b2b5:9999::1/64"
+LXD_IPV6_NAT="true"
+
+# Proxy server
+LXD_IPV6_PROXY="true"
+`
+	expected := `# WARNING: Don't modify this file by hand, it is generated by debconf!
+# To update those values, please run "dpkg-reconfigure lxd"
+
+# Whether to setup a new bridge
+USE_LXD_BRIDGE="true"
+EXISTING_BRIDGE=""
+
+# Bridge name
+LXD_BRIDGE="lxdbr0"
+
+# dnsmasq configuration path
+LXD_CONFILE=""
+
+# dnsmasq domain
+LXD_DOMAIN="lxd"
+
+# IPv4
+LXD_IPV4_ADDR="10.0.19.1"
+LXD_IPV4_NETMASK="255.255.255.0"
+LXD_IPV4_NETWORK="10.0.19.1/24"
+LXD_IPV4_DHCP_RANGE="10.0.19.2,10.0.19.254"
+LXD_IPV4_DHCP_MAX="253"
+LXD_IPV4_NAT="true"
+
+# IPv6
+LXD_IPV6_ADDR="2001:470:b2b5:9999::1"
+LXD_IPV6_MASK="64"
+LXD_IPV6_NETWORK="2001:470:b2b5:9999::1/64"
+LXD_IPV6_NAT="true"
+
+# Proxy server
+LXD_IPV6_PROXY="false"
+
+`
+	result := editLXDBridgeFile(input, "19")
+	c.Assert(result, jc.DeepEquals, expected)
+}
+
+func (s *InitialiserSuite) TestDetectSubnet(c *gc.C) {
+	input := `    inet 127.0.0.1/8 scope host lo
+    inet6 ::1/128 scope host 
+    inet 192.168.0.69/24 brd 192.168.0.255 scope global eth0
+    inet6 fd5d:e5bb:c5f9::c0c/128 scope global dynamic 
+    inet6 fd5d:e5bb:c5f9:0:1e6f:65ff:fed5:5698/64 scope global noprefixroute dynamic 
+    inet6 fe80::1e6f:65ff:fed5:5698/64 scope link 
+    inet 192.168.122.1/24 brd 192.168.122.255 scope global virbr0
+    inet 192.168.100.1/24 brd 192.168.100.255 scope global virbr1
+    inet 10.0.3.1/24 scope global lxcbr0
+    inet6 fe80::a00f:35ff:fe81:f7ed/64 scope link 
+    inet6 fe80::fcd3:9dff:fee4:ba90/64 scope link 
+    inet6 fe80::fc54:ff:fe6e:2d7d/64 scope link 
+    inet6 fe80::fc54:ff:fe3e:8018/64 scope link 
+    inet6 fe80::fc54:ff:feee:c795/64 scope link 
+    inet6 fe80::fc54:ff:fe30:9216/64 scope link 
+`
+	result, err := detectSubnet(input)
+	c.Assert(err, jc.ErrorIsNil)
+	c.Assert(result, jc.DeepEquals, "4")
+}
```

Comment:
Oh, and maybe a test that runs on the active networking configuration. Something that doesn't assert the actual value that we would use, but just something that shows that it is actually an integer, and is in a valid range (0-255).
That should end up getting run on all the platforms we support (we may need to have a SkipIfWindows?), so that we can be fairly covered that we won't just puke on a given platform that we weren't expecting (ppc64 w/ wily or something)